src/core/toga/widgets/button.py

```diff
@@ -27,6 +27,9 @@ def __init__(self, label, id=None, style=None, on_press=None, enabled=True, fact
         self.label = label
         self.on_press = on_press
 
+        # enable or disable the button as the case may be
+        self.enabled = enabled
```

Comment:
This feels like something that should be handled at the base widget level, not on a per-widget level...

Reply:
@freakboy3742 as it can be seen, the _impl variable is being set in the child class after the base constructor has been called. This is the reason I cannot disable the button in the base class's constructor, because when the base class's constructor is being called, _impl  is *none* . This is the reason I am unable to move the above code to the base widget class. One way is that I create another method in the base class and call it from the child class with necessary parameters, once the child has set the _impl variable. The new method serve the purpose of setting up the basic config of a widget. Is that acceptable?